rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -916,10 +918,28 @@ static void handle_builtintopic_endpoint(
       rmw_gid_t ppgid;
       dds_qos_to_rmw_qos(s->qos, &qos_profile);
       convert_guid_to_gid(s->participant_key, ppgid);
+
+      rosidl_type_hash_t type_hash = rosidl_get_zero_initialized_type_hash();
+      void * userdata;
+      size_t userdata_size;
+      if (dds_qget_userdata(s->qos, &userdata, &userdata_size)) {
```

Comment:
I'm pretty sure we are leaking the `userdata` here.  At least, the implementation in https://github.com/eclipse-cyclonedds/cyclonedds/blob/11b923aceb2faa506459fee6cfd708345569ac43/src/core/ddsc/src/dds_qos.c#L31 (which is what `dds_qget_userdata` calls) seems to allocate memory, and we don't seem to be freeing it anywhere.

Reply:
Done! Rebased (all 4 prs) and added `RCPPUTILS_SCOPE_EXIT(dds_free(userdata));` in the if block.